### PR TITLE
Support for full PHPUnit public API

### PR DESCRIPTION
**Pull request by ghost**
- [x] Support for full PHPUnit public API.
- [x] Separate methods inherited exactly from the PHPUnit library to an individual trait for clarity and maintainability.
- [x] Added documentation via PHP annotations.

### CODE REVIEW

**Comment by Naktibalda** [comment]
It is difficult to see what methods were added, because you refactored in the same commit.
Could you provide a list of methods that were added?

**Comment by ghost** [comment]
Done! 😃 You can see the changes [here.](https://github.com/Codeception/lib-asserts/pull/5/commits/d094eba23eab0cd4dd3f83118e0b0b48f813307d)

All were added assertions.
The only two assertions that I took the liberty of removing were:
```
assertArraySubset
assertInternalType
```
Because they were deprecated and eliminated.

@Naktibalda I'll be very aware of your comments. Cheers!

File: src/Codeception/Util/Shared/Asserts.php [hunk 1]
```diff
@@ -189,466 +94,27 @@ protected function assertNotRegExp($pattern, $string, $message = '')
...
-     * @param $message
-     */
-    protected function fail($message)
-    {
-        Assert::fail($message);
-    }
-
-    protected function assertStringContainsString($needle, $haystack, $message = '')
-    {
-        TestCase::assertStringContainsString($needle, $haystack, $message);
```
Comment (Naktibalda) [inline src/Codeception/Util/Shared/Asserts.php:491]: Right, you removed my wrapper methods here, so they are broken with older versions of PHPUnit.

**Comment by brainformatik** [comment]
> Done!  You can see the changes [here.](https://github.com/Codeception/lib-asserts/pull/5/commits/d094eba23eab0cd4dd3f83118e0b0b48f813307d)
> 
> All were added assertions.
> The only two assertions that I took the liberty of removing were:
> 
> ```
> assertArraySubset
> assertInternalType
> ```
> 
> Because they were deprecated and eliminated.
> 
> @Naktibalda I'll be very aware of your comments. Cheers!

Info: we use Codeception 4.0.x with PHPUnit 7.5.x, because of dependency in codeception ^1.0 of this package and the removal of the two mentioned assertions, pipeline fails now.

Because we don't have the time to upgrade to PHPUnit at the moment I think we have to pin to version 1.12.x now.

**Comment by Naktibalda** [comment]
@brainformatik that issue was fixed in lib-asserts 1.13.1, you can upgrade it now.

**Comment by brainformatik** [comment]
@Naktibalda thanks for quick response, I saw the updated version, but error persists:

`PHP Fatal error:  The modifiers of the trait method assertInternalType() are changed, but this method does not exist. Error in ...codeception/module-asserts/src/Codeception/Module/Asserts.php on line 11`

Error comes from module-asserts which we use.

**Comment by Naktibalda** [comment]
I released 1.3.0 of module-asserts too.

**Comment by brainformatik** [comment]
Thanks, with that it's working again.

File: src/Codeception/Util/Shared/Asserts.php (hunk 1 above)
Comment (ghost) [reply src/Codeception/Util/Shared/Asserts.php:491]: Sorry, It wasn't my intention, but I'm glad it could be fixed, thanks you

**Pull request merged**